### PR TITLE
Food vouchers complement in MX

### PR DESCRIPTION
**Pull request by cavalle**
* Adds Food Vouchers complement to MX regime used to generate the CFDI's "ValesDeDespensa" complement.
* Related to the Fuel Account Balance complement (https://github.com/invopop/gobl/pull/210)
* The GOBL complements maps one by one to every node and field of the CFDI complement.
* Implements validations according to the constrains of the the ValesDeDespensa spec.
* Normalises all the amounts to the precision (2 decimals) specified in the ValesDeDespensa spec.
* Calculates the grand total.

### CODE REVIEW

File: regimes/mx/food_vouchers_complement.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	// Date and time of the food voucher's issue (maps to `Fecha`).
+	IssueDateTime cal.DateTime `json:"issue_date_time" jsonschema:"title=Issue Date and Time"`
+	// Employee's tax identity code (maps to `rfc`).
+	EmployeeTaxCode cbc.Code `json:"employee_tax_code" jsonschema:"title=Employee's Tax Identity Code"`
+	// Employee's CURP ("Clave Única de Registro de Población", maps to `curp`).
+	EmployeeCURP cbc.Code `json:"employee_curp" jsonschema:"title=Employee's CURP"`
+	// Employee's name (maps to `nombre`).
+	EmployeeName string `json:"employee_name" jsonschema:"title=Employee's Name"`
+	// Employee's Social Security Number (maps to `numSeguridadSocial`).
+	EmployeeSocialSecurity cbc.Code `json:"employee_social_security,omitempty" jsonschema:"title=Employee's Social Security Number"`
```
Comment (samlown) [inline regimes/mx/food_vouchers_complement.go:60]: Whenever I see namespaces like this I think they could be inside their own object. Would a `FoodVoucherEmployee` object make sense?

Comment (samlown) [reply regimes/mx/food_vouchers_complement.go:60]: I think so... we're renaming everything anyway from Spanish to English, so I do think its reasonable.

Comment (cavalle) [reply regimes/mx/food_vouchers_complement.go]: Cool! Based on [previous feedback](https://github.com/invopop/gobl/pull/192#discussion_r1315048006), I was following the general rule of "stick[ing] as closely as possible to the original source structure, with translated fields". But, I agree that this is a minor deviation from that general rule that makes things nicer for the user. Addressed in 90eaae82147fe06b95f7322ecfb563971bf50cb5.

File: regimes/mx/food_vouchers_complement.go [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+		),
+		validation.Field(&line.IssueDateTime, cal.DateTimeNotZero()),
+		validation.Field(&line.Employee,
+			validation.Required,
+			validation.By(validateFoodVouchersEmployee)),
+		validation.Field(&line.Amount, validation.Required),
+	)
+}
+
+func validateFoodVouchersEmployee(value interface{}) error {
```
Comment (samlown) [inline regimes/mx/food_vouchers_complement.go:109]: BTW, you don't need to manually validate here. You just define the `func (fve *FoodVoucersEmployee) Validate() error` method and ensure the preceding structure includes the field definition:

```
    validation.Field(&line.Employee),
```

Comment (cavalle) [reply regimes/mx/food_vouchers_complement.go]: Nice one! Addressed in cf73168

File: regimes/mx/food_vouchers_complement.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+// Validate checks the FoodVouchersLine data is valid.
+func (fvl *FoodVouchersLine) Validate() error {
+	return validation.ValidateStruct(fvl,
+		validation.Field(&fvl.EWalletID,
+			validation.Required,
+			validation.Length(0, 20),
+		),
+		validation.Field(&fvl.IssueDateTime, cal.DateTimeNotZero()),
+		validation.Field(&fvl.Employee, validation.Required),
```
Comment (samlown) [inline regimes/mx/food_vouchers_complement.go:95]: This looks great now 👍

**Review by samlown — APPROVED** [review]
Looks great! 👍

**Pull request merged**